### PR TITLE
radicle-cli: 0.6.1 -> 0.8.0-unstable-2023-10-11

### PR DESCRIPTION
**Pull request by Luz**
###### Description of changes
- Updated to new repo location
- Using the Cargo.lock from the repo
- Removed many dependencies and tests
- Changed license from gpl3 to MIT

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/radicle-cli/default.nix [hunk 1]
```diff
@@ -2,86 +2,38 @@
...
-    openssh
-  ];
-  preCheck = ''
-    eval $(ssh-agent)
-  '';
-
-  passthru.tests = {
-    version = testers.testVersion { package = radicle-cli; };
-  };
+  doCheck = false;
```
Comment (amesgen) [inline pkgs/applications/version-management/radicle-cli/default.nix:30]: Any specific reason for disabling this? I.e. most tests pass with `nativeCheckInputs = [ git ];`, but there are some failures that might require further investigation.

**Review by amesgen — CHANGES_REQUESTED** [review]
Thanks! A few comments below.

File: pkgs/applications/version-management/radicle-cli/default.nix [hunk 2]
```diff
@@ -2,86 +2,38 @@
...
-  version = "0.6.1";
+  version = "0.8.0";
 
   src = fetchFromGitHub {
     owner = "radicle-dev";
-    repo = pname;
-    rev = "v${version}";
-    sha256 = "sha256-LS6zYpMg0LanRL2M8ioGG8Ys07TPT/3hP7geEGehwxg=";
+    repo = "heartwood";
+    rev = "b27313bc5c703c8a25343176c620c8652b815abc";
```
[the remark this answers is not in the document]
Comment (sebastinez) [reply pkgs/applications/version-management/radicle-cli/default.nix:16]: Hey @amesgen, we currently aren't updating the `version` since we didn't do a proper release of `0.8.0` yet.
Once heartwood is stable we will again follow semver conventions.

I think marking it with `-unstable-` and the date of the fetched commit is a good solution. 👍

File: pkgs/applications/version-management/radicle-cli/default.nix (hunk 1 above)
Comment (sebastinez) [reply pkgs/applications/version-management/radicle-cli/default.nix:30]: Yeah we saw some issues, with running `cargo test` in the `installCheckPhase`.
Opened a topic in the Radicle chat to see if we can debug the errors that require further investigation.
https://radicle.zulipchat.com/#narrow/stream/369277-heartwood/topic/nix.20package/near/376875202

File: pkgs/applications/version-management/radicle-cli/default.nix [hunk 3]
```diff
@@ -2,86 +2,38 @@
...
-    version = testers.testVersion { package = radicle-cli; };
-  };
+  doCheck = false;
 
   meta = {
     description = "Command-line tooling for Radicle, a decentralized code collaboration network";
     homepage = "https://radicle.xyz";
-    license = lib.licenses.gpl3Plus;
-    maintainers = with lib.maintainers; [ amesgen ];
+    license = lib.licenses.mit;
```
[the remark this answers is not in the document]
Comment (gsaslis) [reply pkgs/applications/version-management/radicle-cli/default.nix:35]: good point

https://app.radicle.xyz/seeds/seed.radicle.xyz/rad:z3gqcJUoA1n9HaHKufZs5FCSGazv5 should be the source of truth however ;)

[the remark this answers is not in the document]
Comment (sebastinez) [reply pkgs/applications/version-management/radicle-cli/default.nix:35]: @gsaslis hehe I also told @Luz that but I think for now it's not that bad to follow the binaries that GitHub CI outputs currently.
Once we have CI on Radicle we should change that for sure and use `fetchGit` instead of `fetchFromGitHub`

File: pkgs/applications/version-management/radicle-cli/default.nix (hunk 1 above)
Comment (sebastinez) [reply pkgs/applications/version-management/radicle-cli/default.nix:30]: I think you can eliminate the other failures by adding

```
cargoTestFlags = [
    "--all"
  ];
```

Comment (sebastinez) [reply pkgs/applications/version-management/radicle-cli/default.nix:30]: PS. I think we should also add `cargoBuildFlags = ["--workspace"];` so we get `radicle-httpd` and `radicle-tui` too

File: pkgs/applications/version-management/radicle-cli/default.nix [hunk 4]
```diff
@@ -2,86 +2,38 @@
...
-  buildInputs = [
-    openssl
-  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
-    libusb1
+  buildInputs = [ ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
     AppKit
   ];
 
-  postInstall = ''
-    for f in $(find . -name '*.adoc'); do
```
Comment (sebastinez) [inline pkgs/applications/version-management/radicle-cli/default.nix:61]: Maybe we can revert this change since we have now again .adoc files.
Thanks to this commit

https://github.com/radicle-dev/heartwood/commit/af28a0ca07c2a0601e39b14fe3054ecb3f9a3bbc

File: pkgs/applications/version-management/radicle-cli/default.nix (hunk 1 above)
Comment (amesgen) [reply pkgs/applications/version-management/radicle-cli/default.nix:30]: Progress: much more tests pass with
```nix
  cargoTestFlags = [ "--workspace" ];
  checkType = "debug";
  nativeCheckInputs = [ git tree ];
```
But there are still some failures left...

File: pkgs/applications/version-management/radicle-cli/default.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (Luz) [reply pkgs/applications/version-management/radicle-cli/default.nix]: If i understand this correctly, we could do two steps:
1) Change to `fetchGit` and use the git repo on radicle's server: https://seed.radicle.xyz/z3gqcJUoA1n9HaHKufZs5FCSGazv5.git
2) Change to `fetchRadicle` (which probably doesnt exist yet) and use the source of truth according gsaslis
Should I do "1)"?

File: pkgs/applications/version-management/radicle-cli/default.nix (hunk 1 above)
Comment (Luz) [reply pkgs/applications/version-management/radicle-cli/default.nix:40]: Thanks at all for your time and the awesome suggestions. I've tried to combine all inputs into a fresh commit.
However, in case I activate the tests, I cannot get it to succeed.
Latest error message I get is something with `rad_sync`, see below. Might be, because I removed some dependencies?
```
       > failures:
       >     rad_sync
       >
       > test result: FAILED. 33 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in
```

File: pkgs/applications/version-management/radicle-cli/default.nix [hunk 5]
```diff
@@ -2,86 +2,38 @@
...
-    git
-    openssh
-  ];
-  preCheck = ''
-    eval $(ssh-agent)
-  '';
-
-  passthru.tests = {
-    version = testers.testVersion { package = radicle-cli; };
-  };
```
[the remark this answers is not in the document]
Comment (Luz) [reply pkgs/applications/version-management/radicle-cli/default.nix:78]: What does this do? :)

File: pkgs/applications/version-management/radicle-cli/default.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (sebastinez) [reply pkgs/applications/version-management/radicle-cli/default.nix]: I think if we as next step, maybe once Radicle moves CI to it's own solution, move to `fetchGit` that should already be enough.
I don't think we need a `fetchRadicle` fn since Radicle is built on top of git.

File: pkgs/applications/version-management/radicle-cli/default.nix (hunk 5 above)
[the remark this answers is not in the document]
Comment (amesgen) [reply pkgs/applications/version-management/radicle-cli/default.nix:78]: It tests that the main binary reports a version that is in sync with what is specified in Nix: https://nixos.org/manual/nixpkgs/unstable/#tester-testVersion
Concretely, you can this snippet here:
```nix
  passthru.tests = {
    version = testers.testVersion {
      package = radicle-cli;
      version = lib.head (lib.splitString "-" version);
    };
  };
```

File: pkgs/applications/version-management/radicle-cli/default.nix (hunk 1 above)
Comment (Luz) [reply pkgs/applications/version-management/radicle-cli/default.nix:40]: thread 'rad_sync' panicked at 'Node::routes_to: unexpected route for ...
... radicle-cli/tests/commands.rs:792:9

Comment (Luz) [reply pkgs/applications/version-management/radicle-cli/default.nix:40]: Fixed by removing the race conditions (of the multiple threads):
`dontUseCargoParallelTests = true`

File: pkgs/applications/version-management/radicle-cli/default.nix [hunk 6]
```diff
@@ -2,61 +2,58 @@
...
-    };
-  };
+  cargoHash = "sha256-69qdH7qlXS0zK9lFPbl0IPmh01N/aIseGLBL1BNAtP0=";
 
-  # Otherwise, there are errors due to the `abigen` macro from `ethers`.
-  auditable = false;
+  patches = [
+    ./ignore-test-with-race-condition.patch
+    ./ignore-tests-requiring-terminal.patch
+  ];
```
Comment (amesgen) [inline pkgs/applications/version-management/radicle-cli/default.nix:31]: Thanks for your work on getting the tests to pass! It seems that we still have failures on `x86_64-darwin`: https://github.com/NixOS/nixpkgs/pull/244417/checks?check_run_id=15408058168

In general, I think it is acceptable to disable the tests (maybe just on non-`x86_64-linux`) if it turns out too difficult to get them to pass; especially as radicle is currently still on an experimental version.

**Comment by Luz** [comment]
@lunaticare Maybe you could test this on aarch64-darwin?

**Comment by lunaticare** [comment]
> @lunaticare Maybe you could test this on aarch64-darwin?

@Luz Test are failing:
```
thread 'tests::e2e::test_clone' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Error { kind: InvalidInput, message: "path must be shorter than SUN_LEN" })', radicle-node/src/test/environment.rs:371:10
```

Full build log https://pastebin.com/pwhnS2fS

**Comment by lunaticare** [comment]
@Luz aarch64-darwin build succeeded on https://github.com/NixOS/nixpkgs/commit/00399959647aba185542a146383508f4789180b9

**Review by amesgen — APPROVED** [review]
Can confirm that everything now builds/tests fine on `{x86_64,aarch64}-{linux,darwin}` :+1:

**Comment by lunaticare** [comment]
@Luz nixpkgs' current version of Radicle does not work with the current network, and is as good for practical purposes as having no package at all. Do you think it's convenient to merge the package definition now so people can start using it, and fix tests and packaging issues later?

**Comment by Luz** [comment]
> @Luz nixpkgs' current version of Radicle does not work with the current network, and is as good for practical purposes as having no package at all. Do you think it's convenient to merge the package definition now so people can start using it, and fix tests and remaining packaging issues later?

Yes, I would definately merge it now, imo :)

File: pkgs/applications/version-management/radicle-cli/default.nix [hunk 7]
```diff
@@ -2,61 +2,56 @@
...
-  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
+  nativeBuildInputs = [ asciidoctor installShellFiles ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
     DarwinTools
   ];
 
-  buildInputs = [
-    openssl
-  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
-    libusb1
+  buildInputs = [ ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/radicle-cli/default.nix:35]:
```suggestion
  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [
```

**Comment by Luz** [comment]
> @Luz Is this proposed state still working?

I've just heard that there is soon a release of version 1.0, so we probably just wait for 1.0 and then build that version

**Comment by Luz** [comment]
I'll close this, best is to wait for version 1.0 and then build that version.
